event_reporter/utils/app_revision.go

```diff
@@ -65,14 +72,38 @@ func GetOperationRevision(a *appv1.Application) string {
 	return revision
 }
 
-func GetOperationStateRevision(a *appv1.Application) *string {
+func GetOperationRevisions(a *appv1.Application) []string {
+	if a == nil {
+		return nil
+	}
+
+	// this value will be used in case if application hasn't resources , like gitsource
+	revisions := a.Status.Sync.Revisions
+	if a.Status.OperationState != nil && a.Status.OperationState.Operation.Sync != nil && a.Status.OperationState.Operation.Sync.Revisions != nil && len(a.Status.OperationState.Operation.Sync.Revisions) > 0 {
```

Comment:
lets change to if -> return, and tbh i would move these if condition to some function, it is just very hard to read, i would move it to our custom types that we created before

Reply:
moved to separate func